algorithms/closest_pair.py

```diff
@@ -84,7 +84,12 @@ def _distance_between(point_1, point_2):
 
```

Comment:
Have just tried the algorithm with the inputs: `[(0, 2),(2, 2),(3, 2),(5, 2),(7, 2),(9, 2),(11, 2),(14, 2)]`.
Clearly `(2, 2)` and `(3, 2)` are the closest, but the algorithm spits out `(3, 2), (5, 2)`. I think because each recursive call doesn't also work out the closest split pair...